### PR TITLE
W-16571553: merge avro feature to develop in amf

### PR DESCRIPTION
**Pull request by arielmirra**
- **W-16459888: add AvroRawSchema Annotation**
- **W-16459888: add Avro JVM and JS plugins**
- **W-16459888: split avro/jsonschema validation logic**
- **W-16459888: initial validation testing**
- **W-16540082: trow JVM-specific validation messages**
- **W-16540082: simplify avro payload validators**
- **W-16540082: improve JS avro payload validation**
- **W-15633281: add JVM schema validation tests**
- **W-15633281: add JS AVRO Schema validation dummy code**
- **W-15633281: fix avro plugin validation error message**
- **W-16540082: stop serializing AVRORawSchema annotation**
- **W-16609870: make async message payload object be virtual unless explicitly declared**
- **return an empty payload when parsing invalid avro**
- **W-16540082: rename avro Document Emitter**
- **W-16540082: fix Async AVRO payload emission**
- **W-16540082: improve AVRO adr doc**
- **W-16540082: improve base avro parser interfaces**
- **W-16540082: improve avro validation exceptions in JVM**
- **W-16540082: make AVRORawSchema ann. Eternal**
- **W-16540082: add YAML to avro validator mediaTypes**
- **add definedBySpec annotation to avro schemas in async apis**
- **W-16596042: fix avro map with empty values lexical**
- **W-16596042: update avro ADR validation constraints**
- **W-16701554: test avro default values validations**
- **W-16702592 - AVRO Js lib working**
- **generalize loadDataNodeString method in avro validators**
- **fix avro collection members parsing for ALS**
- **W-16701643: add AvroSchemaPayloadValidationPlugin**
- **W-16702592 - Extracted AVRO Error filter. Fixed invalid root type validation**
- **W-16701643: remove duplicated anns. in avro record range fields**
- **W-16702592 - Fix schema parse validation. Remove excluded errors (we have single error, is not possible to skip)**
- **W-16702592 - Uncomment some tests that was ignored for JS**
- **W-16701643: put avro record field default value inside the property range**
- **W-16701643: test async+avro payload validation in channel message examples**
- **W-16702592 - Added custom NameValidator to JVM AVRO parser in order to support some special chars in names**
- **W-16701643: test async+avro payload validator creation**
- **update avro ADR**
- **W-16701643: add anns. to empty avro type**

### CODE REVIEW

File: adrs/0014-avro-support.md [hunk 1]
```diff
@@ -50,10 +52,32 @@ We've also added 3 AVRO-specific fields to the `AnyShape` Model via the `AvroFie
 - Aliases
 - Size
 
+### Where we support and DON'T support AVRO Schemas
+We Support AVRO Schemas (inline or inside a `$ref`):
+- as a standalone document or file 
+  - we encourage users to use the `.avsc` file type to indicate that's an avro file, for better suggestions and so on in the platform).
+- inside a message payload in an AsyncAPI
+  - the key `schemaFormat` MUST be declared and specify it's an AVRO payload
```
Comment (looseale) [inline adrs/0014-avro-support.md:61]: I would add that avro specific document (AvroSchemaDocument) could be only parsed with the specific AvroConfiguration

File: adrs/0014-avro-support.md [hunk 2]
```diff
@@ -50,10 +52,32 @@ We've also added 3 AVRO-specific fields to the `AnyShape` Model via the `AvroFie
...
+### JVM avro validation plugin
+- validation per se is not supported, we try to parse an avro schema and throw parsing results if there are any
+  - this means it's difficult to have location of where the error is thrown, we may give an approximate location from our end post-validation
+- when a validation is thrown, the rest of the file is not being searched for more validations
+  - this is particularly important in large avro schemas, where many errors can be found but only one is shown
+
+
+### Both JVM & JS validation plugins
+- `"default"` values are not being validated when the type is `bytes`, `map`, or `array`
+- the validator treats as invalid an empty array as the default value for arrays (`"default": []`) even though the [Avro Schema Specification](https://avro.apache.org/docs/1.12.0/specification) has some examples with it
```
Comment (looseale) [inline adrs/0014-avro-support.md:84]: I would add the limitation in validating default and payload in the Shape associated to a field in a record that has a recursive reference

Comment (looseale) [reply adrs/0014-avro-support.md:84]: Adding that now we are failing, in the future we will try to ignore this cases and show a warning

File: adrs/0014-avro-support.md [hunk 3]
```diff
@@ -50,10 +52,32 @@ We've also added 3 AVRO-specific fields to the `AnyShape` Model via the `AvroFie
...
+- inside a message payload in an AsyncAPI
+  - the key `schemaFormat` MUST be declared and specify it's an AVRO payload
 
-For now only parsing is done 
+We don't support AVRO Schemas:
+- inside components --> schemas in an AsyncAPI
+  - because we can't determine if it's an AVRO Schema or any other schema
 
+### AVRO Validation
+We'll use the Apache official plugins for JVM and JS.
```
Comment (looseale) [inline adrs/0014-avro-support.md]: libraries, not plugins

**Review by looseale — APPROVED** [review]

**Review by jisoldi — APPROVED** [review]

**Pull request merged**